reactor-netty-http-brave/src/main/java/reactor/netty/http/client/BraveHttpClientTracing.java

```diff
@@ -0,0 +1,291 @@
+			                  else if (state == RESPONSE_INCOMPLETE) {
+			                      Throwable error = new PrematureCloseException("Connection prematurely closed DURING response");
+			                      handleReceive((AtomicReference<Span>) tracingContext, ops, request, error);
+			                  }
+			              });
+		}
+	}
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public void onUncaughtException(Connection connection, Throwable error) {
+		if (connection instanceof HttpClientOperations) {
+			HttpClientOperations ops = (HttpClientOperations) connection;
+			TracingContext.of(ops.currentContext())
+			              .ifPresent(tracingContext -> {
+			                  if (error instanceof RedirectClientException) {
+			                      handleReceive((AtomicReference<Span>) tracingContext,
+			                              ops, new DelegatingHttpRequest(ops, uriMapping), null);
+			                  }
+			                  else if (ops.retrying && ops.responseState == null) {
+			                      handleReceive((AtomicReference<Span>) tracingContext,
+			                              ops, new DelegatingHttpRequest(ops, uriMapping), error);
+			                  }
+			                  else if (ops.responseState != null) {
+			                      handleReceive((AtomicReference<Span>) tracingContext,
+			                              ops, new DelegatingHttpRequest(ops, uriMapping), error);
+			                  }
+			              });
+		}
+	}
+
+	void handleReceive(AtomicReference<Span> tracingContext, HttpClientOperations ops, HttpClientRequest request, @Nullable Throwable throwable) {
+		Span span = tracingContext.getAndSet(null);
+		if (span != null) {
+			handler.handleReceive(new DelegatingHttpResponse(ops, request, throwable), span);
+		}
+	}
+
+	final CurrentTraceContext currentTraceContext;
+	final HttpClientHandler<HttpClientRequest, HttpClientResponse> handler;
+	final Function<String, String> uriMapping;
+
+	BraveHttpClientTracing(HttpTracing httpTracing, Function<String, String> uriMapping) {
+		requireNonNull(httpTracing, "httpTracing");
+		this.currentTraceContext = httpTracing.tracing().currentTraceContext();
+		this.handler = HttpClientHandler.create(httpTracing);
+		this.uriMapping = requireNonNull(uriMapping, "uriMapping");
+	}
+
+	@Override
+	public Mono<? extends Connection> apply(Mono<? extends Connection> connectionMono) {
+		BraveHttpClientMapConnect  mapConnect = new BraveHttpClientMapConnect();
+		return connectionMono.doOnError(error -> {
+		                         Span span = mapConnect.getAndSet(null);
+		                         if (span != null) {
+		                             span.error(error).finish();
+		                         }
+		                     })
+		                     .doOnCancel(() -> {
+		                         Span span = mapConnect.getAndSet(null);
+		                         if (span != null) {
+		                             span.annotate("cancel").finish();
```

Comment:
one thing I think I missed in the previous PR. consider extracting to a shared constant (and reuse in `BraveHttpServerTracing`)

Reply:
I might miss something but `mapConnect` cannot be used in the server.